pkgcore/operations/format.py

```diff
@@ -45,10 +45,11 @@ def fetch_one(self, fetchable, observer):
         # XXX hack atm, could use better logic but works for now
         try:
             fp = self.fetcher(fetchable)
-        except Exception:
+        except Exception as e:
+            observer.error(str(e))
+            observer.error("failed fetching %s", fetchable)
             fp = None
         if fp is None:
-            self.failed_fetch(fetchable, observer)
```

Comment:
I made it work ~3 years ago with 76747f. If you're talking about forgetting to only run it for fetch restricted packages as per the spec, then yes I just noticed and fixed that. :smile: